SwiftKit/src/main/java/org/swift/swiftkit/SwiftInstanceCleanup.java

```diff
@@ -22,51 +22,6 @@
 interface SwiftInstanceCleanup extends Runnable {
 }
 
-/**
- * Implements cleaning up a Swift {@link SwiftHeapObject}.
- * <p>
- * This class does not store references to the Java wrapper class, and therefore the wrapper may be subject to GC,
- * which may trigger a cleanup (using this class), which will clean up its underlying native memory resource.
- */
-// non-final for testing
-class SwiftHeapObjectCleanup implements SwiftInstanceCleanup {
-
-    private final MemorySegment selfPointer;
-    private final SwiftAnyType selfType;
-    private final Runnable markAsDestroyed;
-
-    /**
-     * This constructor on purpose does not just take a {@link SwiftHeapObject} in order to make it very
-     * clear that it does not take ownership of it, but we ONLY manage the native resource here.
-     * </p>
-     * This is important for {@link AutoSwiftMemorySession} which relies on the wrapper type to be GC-able,
-     * when no longer "in use" on the Java side.
-     */
-    SwiftHeapObjectCleanup(MemorySegment selfPointer,
-                           SwiftAnyType selfType, Runnable markAsDestroyed) {
-        this.selfPointer  = selfPointer;
-        this.markAsDestroyed = markAsDestroyed;
-        this.selfType = selfType;
-    }
-
-    @Override
-    public void run() throws UnexpectedRetainCountException {
-        // Verify we're only destroying an object that's indeed not retained by anyone else:
-        long retainedCount = SwiftKit.retainCount(selfPointer);
-        if (retainedCount > 1) {
-            throw new UnexpectedRetainCountException(selfPointer, retainedCount, 1);
-        }
```

Comment:
Because `SwiftHeapObjectCleanup` is now NOT used anymore, `arena_releaseClassOnClose_class_leaked` test is now failing, but this check is not okay because somewhere in the Swift world may still holds the instance.